src/main/java/pl/edu/mimuw/Program.java

```diff
@@ -0,0 +1,41 @@
+package pl.edu.mimuw;
+
+import java.util.ArrayList;
+
+public class Program {
+  private String name;
+  private String programType;
+  private ArrayList<Subject> obligatory;
+  private ArrayList<Subject> facultative;
+
+  public Program() {
+  }
+
+  public Program(String name, String programType) {
+    this.name = name;
+    this.programType = programType;
+  }
+
+  public Program(String name, String programType, ArrayList<Subject> obligatory, ArrayList<Subject> facultative) {
+    this(name, programType);
+    this.obligatory = obligatory;
+    this.facultative = facultative;
+  }
+
+  public String toString(boolean withSubjects) {
```

Comment:
to jest myląca technika - lepiej chyba w takich przypadkach nazywać metody `createRepresentation` etc. bo za bardzo wszyscy są przyzwyczajeni, czym jest `toString`